### DISCUSSION TITLE
switch to interactive mode from cli

### DISCUSSION THREAD

**Discussion by InfantileMan** [daniele77/cli#159] [Q&A] [answered]
Hi, Daniele
Is there any way to switch to interactive mode from cli?
For example I have function UploadToDevice() which searches for connected usb devices and show them by list:
Connected usb-devices:
1) first device
2) second device
3) third device
and then asks the user which one to choose. User should enter the number 1-3, but the user does not see the input number because terminal in "ManualMode" then program requires confirmation from user and user must again enter something and etc.
I call function UploadToDevice() inside lambda
```
rootMenu->Insert("Upload", [](std::ostream& out){
  ...
  UploadToDevice();
  ...
});
```
user input inside UploadToDevice function should not be written to history, but user should see their input in terminal
how to solve this problem?

**Comment by daniele77** [comment, marked as answer]
Unfortunately, I think it's not possible to mix up CLI with other functions that wait for user input.
You can try by using `CliFileSession `instead of `CliLocalSession` (see example `simplelocalsession.cpp`): in that way you don't get advanced features of cli (e.g., history and autocompletion), but maybe you can integrate your function easily.

**Reply by InfantileMan** [reply]
Daniele, thanks for the answer and with the existing functionality the library is pretty good.
